### PR TITLE
chore: update Dockerfile to reduce the image size

### PR DESCRIPTION
**Pull request by tdurieux**
Hi there,

I've made a small improvement to the Dockerfile that I think could help optimize the image size.

Summary of the changes:
* I added `npm cache clean` after `npm install` to remove `npm` cache that is not needed inside the Docker image.
* I added the `--no-cache-dir` to the `pip` command to disable the package cache.
* I added the `--no-install-recommends` to with apt-get in order to not install unnecessary packages and reduce the image size.

Impact on the image size:
* Image size before repair: 1.28 GB
* Image size after repair: 1.15 GB
* Difference: 132.96 MB (10.15%)

I hope that you will find these changes useful to you. Let me know if you have any questions or concerns.

Thanks,

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @tdurieux!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.


:warning: We can't start reviewing your pull request until you've submitted a     [signed contributor agreement](https://openedx.org/cla)     or indicated your institutional affiliation. Please see the [CONTRIBUTING](https://github.com/openedx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.     If you've signed an agreement in the past, you may need to re-sign.  See     [The New Home of the Open edX Codebase](https://open.edx.org/blog/the-new-home-of-the-open-edx-codebase/)     for details.

    
Once you've signed the CLA, please allow 1 business day for it to be processed.  After this time, you can re-run the     CLA check by editing the PR title.  If the problem persists, you can tag the `@openedx/cla-problems` team in a     comment on your PR for further assistance.

**Comment by mphilbrick211** [comment]
Hi @tdurieux! Thank you for this contribution! In order to proceed, you'll need to fill out a CLA form as mentioned above. Please let me know if you have any questions. Thanks!

**Comment by tdurieux** [comment]
I just sign it, thank you for the reminder @mphilbrick211!

**Comment by mphilbrick211** [comment]
Hi @tdurieux! It looks like your CLA may not have gone through. If it was processed correctly, your CLA check would turn green when re-running tests.

Please note: We ran into some issues on our end getting CLAs processed over the last week or so, so you may need to submit again. Sorry for the inconvenience.

**Comment by mphilbrick211** [comment]
> Hi @tdurieux! It looks like your CLA may not have gone through. If it was processed correctly, your CLA check would turn green when re-running tests.
> 
> Please note: We ran into some issues on our end getting CLAs processed over the last week or so, so you may need to submit again. Sorry for the inconvenience.

Hi @tdurieux! Friendly check-in to see if you plan to pursue this pull request? If so, per the message above, you'll need to make sure you submit a CLA form. If you have already done so, when you re-run the tests the CLA check will appear green. Please let me know if you have any questions. Thanks!

**Comment by mphilbrick211** [comment]
Hi @tdurieux - following up to see if you plan to pursue this pull request?

**Comment by tdurieux** [comment]
I do, I have a lot of work for the moment. I will come back to it today or tomorrow

**Comment by tdurieux** [comment]
I submitted again my CLA

**Comment by mphilbrick211** [comment]
> I submitted again my CLA

Thanks, @tdurieux! If you re-run the tests, the CLA check should turn green.

**Comment by tdurieux** [comment]
I restarted the tests, everything is green except codecov/project but there is nothing I can do about that.

**Comment by mphilbrick211** [comment]
Thanks @tdurieux! We'll get this reviewed for you.

**Comment by mphilbrick211** [comment]
Hi @tdurieux! Would you mind re-running the tests - I think the codecov failure should be all set now.

@openedx/2u-aperture would someone be able to review this? Thanks!

**Comment by mphilbrick211** [comment]
> Hi @tdurieux! Would you mind re-running the tests - I think the codecov failure should be all set now.
> 
> @openedx/2u-aperture would someone be able to review this? Thanks!

Hi @tdurieux - friendly follow-up on this!

**Comment by tdurieux** [comment]
I am not able to restart the tests since I am not inside your organization. You should ask one of your dev to do it.

**Comment by mphilbrick211** [comment]
Hi @openedx/2u-aperture - could someone help with this one? Thanks!

**Comment by jsnwesson** [comment]
Hi all, I know this has been hanging around for a while, an internal Jira ticket has been created and I'll be bringing it up in tomorrow's team meeting.

**Comment by justinhynes** [comment]
We didn't get to prioritizing this in our recent refinement meetings, I just bumped this ticket to the top of the queue for our next session, which will happen on Tuesday next week.

File: Dockerfile [hunk 1]
```diff
@@ -33,9 +33,9 @@ MAINTAINER devops@edx.org
 
 # If you add a package here please include a comment above describing what it is used for
 RUN apt-get update && \
-apt-get install -y software-properties-common && \
+ apt-get install --no-install-recommends -y software-properties-common && \
```
Comment (deborahgu) [inline Dockerfile:36]: did you mean to add the leading whitespace here?

**Review by deborahgu — COMMENTED** [review]
this looks good to me, except for an errant whitespace. It has some conflicts that need to be resolvedbut presumably those will go in cleanly..

**Review by deborahgu — CHANGES_REQUESTED** [review]
clean up the white space and do the conflict resolution and I think it should be fine

**Comment by tdurieux** [comment]
@deborahgu I made the requested changes.

**Review by deborahgu — APPROVED** [review]
thanks!

**Comment by openedx-webhooks** [comment]
@tdurieux 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/credentials&entry.867055334=https://github.com/openedx/credentials/pull/1912&entry.1484655318=https://github.com/tdurieux&entry.752974735=2023-01-22+20:32&entry.1917517419=2023-10-02+15:36&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**